README.md

```diff
@@ -19,6 +19,25 @@ It contains *no compiled code* meaning that it's *easy to install* and
 use on all platforms that support NumPy.  Despite being fully interpreted, the
 code has a reasonably *good performance*.
 
+The following benchmark (`docs/examples/performance.py`) demonstrates how the
+time spent in finite element assembly is less than the time spent in linear
+solve for larger problems.  The benchmark uses linear tetrahedral elements and
+the default direct sparse solver of `scipy.sparse.linalg.spsolve` for solving
+the Laplace equation.
+
+| Degrees-of-freedom | Time spent in assembly (s) | Time spent in linear solve (s) |
+| --- | --- | --- |
+| 1000 | 0.020067664658805977 | 0.0037054363444137075 |
+| 1728 | 0.022763118671718985 | 0.008123454657228043 |
+| 2744 | 0.03378102233788619 | 0.020783308330768097 |
+| 4096 | 0.044692154663304486 | 0.05756178432299445 |
+| 5832 | 0.06529411301016808 | 0.26006614298482117 |
+| 8000 | 0.09247205066882695 | 0.41909634032829973 |
+| 10648 | 0.14258869632612914 | 1.0872577403400403 |
+| 13824 | 0.1709109363340152 | 2.1439098239837526 |
+| 17576 | 0.23109773001245534 | 4.2809405889905365 |
+| 21952 | 0.3161410770068566 | 8.190604210676005 |
```

Comment:
I'd have been tempted to go higher here.  I've run ex32 with over three million tetrahedra; the currently demonstrated approaches scale reasonably to that size.

Reply:
Sure, I was more attempting to demonstrate that when increasing the number of DOF's, the linear solve time will eventually dominate.

Reply:
I mean, the linear solve is proportional to O(N^3) whereas the assembly seems to scale about O(N).

Reply:
True, but one isn't limited to direct solution in scikit-fem and it's usual to switch to something iterative for larger problems, generally ultimately aiming for O(_N_).  In the same way that for very small problems one might use dense matrices instead of sparse and for even smaller problems, one might invert with Cramer's rule.

By which I mean that the time to assemble a sparse matrix for a problem with 1e6 degrees of freedom may well be of interest to users even if direct solution is not recommended, and in such a case, the time for direct solution might not be that relevant.

Actually the first ever problem that I solved with scikit-fem had three million P1 triangles and did use the direct solver.  The whole script took five minutes (which I was very pleased with).  The largest part of that was reading the mesh!  That has since been remedied in meshio.

I had a go at extending the results (skipping solution for larger problems) and got:
| Degrees-of-freedom | Time spent in assembly (s) | Time spent in linear solve (s) |
| --- | --- | --- |
| 8 | 0.0028495680003288726 | 0.0015545840001323086 |
| 27 | 0.0037092200000188313 | 0.001653416999943147 |
| 64 | 0.003610274000038771 | 0.005056961999798659 |
| 125 | 0.004651539999940724 | 0.001691012000264891 |
| 216 | 0.007155211999815947 | 0.0018959560002258513 |
| 512 | 0.014114281999809464 | 0.003238801999941643 |
| 1000 | 0.02644422499997745 | 0.007219724000151473 |
| 1728 | 0.046652433999952336 | 0.018336376000206656 |
| 4096 | 0.11173986999983754 | 0.15292731399995318 |
| 8000 | 0.2402442269999483 | 0.9322175259999312 |
| 15625 | 0.49199637799983975 | 6.394490390000101 |
| 32768 | 1.075414314000227 | 41.25685121299966 |
| 64000 | 2.190861079000115 | nan |
| 125000 | 5.05543281700011 | nan |
| 262144 | 12.957759258999886 | nan |
| 512000 | 297.89818344500054 | nan |

This is on my ten-year old commodity laptop; I think it has 4GB of RAM.  It clearly ran into trouble assembling the last one but was pretty close to O(_N_) up to then.  I don't have a bigger machine available today.  I think one could extrapolate to a figure of about a minute to assemble a million DOFs on a MeshTet?  (On this machine.)  That's a nice round estimate for a user to make sense of.

The refinement loop in performance.py was modified to: 
```python
for k in range(4, 20):
    m = pre(int(2**(k/3)))
```

Reply:
I reran that on another machine.  It has 8GB of RAM so made it to a million DOFs, but it's even slower than my laptop.  (It's a Windows 10 laptop that I've borrowed so that I can work from home.)

| Degrees-of-freedom | Time spent in assembly (s) | Time spent in linear solve (s) |
| --- | --- | --- |
| 8 | 0.022863499999999704 | 0.06908090000000033 |
| 27 | 0.011188700000000829 | 0.008434699999999573 |
| 64 | 0.021992299999999965 | 0.09383180000000024 |
| 125 | 0.02371719999999833 | 0.003881400000000923 |
| 216 | 0.03649829999999987 | 0.007528999999999897 |
| 512 | 0.048406399999999294 | 0.010408399999999318 |
| 1000 | 0.09417129999999929 | 0.1059594999999991 |
| 1728 | 0.23135640000000102 | 0.19209790000000027 |
| 4096 | 0.35432750000000013 | 0.2926724000000007 |
| 8000 | 0.5209161000000009 | 1.5828071000000001 |
| 15625 | 1.718046900000001 | 3.7550335999999973 |
| 32768 | 4.464542200000004 | 27.226801700000003 |
| 64000 | 9.470034400000003 | nan |
| 125000 | 17.250329099999988 | nan |
| 262144 | 37.03513559999999 | nan |
| 512000 | 76.08782049999996 | nan |
| 1030301 | 311.34253179999996 | nan |

This seems to hit trouble on the last one too but is roughly O(_N_) up until then.  These old laptops aren't sensible things to be benchmarking with, sorry.  (I am very glad that I can run scikit-fem on them though.)

Reply:
This is on my 7th gen ThinkPad X1 Carbon:

| Degrees-of-freedom | Time spent in assembly (s) | Time spent in linear solve (s) |
| --- | --- | --- |
| 64 | 0.001719430962111801 | 0.0008631939999759197 |
| 125 | 0.0022377190180122852 | 0.0009978360030800104 |
| 216 | 0.004175334994215518 | 0.0016233620117418468 |
| 512 | 0.006831337988842279 | 0.0018000249983742833 |
| 1000 | 0.027329800999723375 | 0.006487983977422118 |
| 1728 | 0.02035172702744603 | 0.00894418201642111 |
| 4096 | 0.04277183400699869 | 0.054088378965388983 |
| 8000 | 0.08925403101602569 | 0.39822074701078236 |
| 15625 | 0.17675437999423593 | 3.5216876140329987 |
| 32768 | 0.41025503899436444 | 24.167141190962866 |
| 64000 | 0.8814217780018225 | 138.6927448490169 |
| 125000 | 1.7890089299762622 | nan |
| 262144 | 4.044992835028097 | nan |
| 512000 | 8.037139102991205 | nan |
| 1030301 | 20.241967056994326 | nan |

Reply:
Cool! I better get me one of them.

Rather better than the neat 'million in a minute'.